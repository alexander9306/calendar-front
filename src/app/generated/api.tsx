import { gql } from '@apollo/client';
import * as Apollo from '@apollo/client';
export type Maybe<T> = T | null;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
  /** A date-time string at UTC, such as 2019-12-03T09:54:33Z, compliant with the date-time format. */
  DateTime: string;
  stringOrBoolean: any;
};

export type CreateEventInput = {
  name: Scalars['String'];
  description?: Maybe<Scalars['String']>;
  date: Scalars['DateTime'];
};


export type Event = {
  __typename?: 'Event';
  id: Scalars['ID'];
  name: Scalars['String'];
  description?: Maybe<Scalars['String']>;
  date: Scalars['DateTime'];
  createdAt: Scalars['DateTime'];
  updatedAt: Scalars['DateTime'];
};

export type Mutation = {
  __typename?: 'Mutation';
  createEvent: Event;
  updateEvent: Event;
  removeEvent: Event;
};


export type MutationCreateEventArgs = {
  createEventInput: CreateEventInput;
};


export type MutationUpdateEventArgs = {
  updateEventInput: UpdateEventInput;
};


export type MutationRemoveEventArgs = {
  id: Scalars['ID'];
};

export type Query = {
  __typename?: 'Query';
  event: Event;
  events: Array<Event>;
  isDarkTheme: Scalars['Boolean'];
  isLoggedIn: Scalars['Boolean'];
  loadingBarStatus: Scalars['Int'];
  sidebarShow: Scalars['stringOrBoolean'];
};


export type QueryEventArgs = {
  id: Scalars['ID'];
};


export type UpdateEventInput = {
  name?: Maybe<Scalars['String']>;
  description?: Maybe<Scalars['String']>;
  date?: Maybe<Scalars['DateTime']>;
  id: Scalars['ID'];
};

export type GetEventsQueryVariables = Exact<{ [key: string]: never; }>;


export type GetEventsQuery = (
  { __typename?: 'Query' }
  & { events: Array<(
    { __typename?: 'Event' }
    & Pick<Event, 'id' | 'name' | 'date'>
  )> }
);


export const GetEventsDocument = gql`
    query GetEvents {
  events {
    id
    name
    date
  }
}
    `;

/**
 * __useGetEventsQuery__
 *
 * To run a query within a React component, call `useGetEventsQuery` and pass it any options that fit your needs.
 * When your component renders, `useGetEventsQuery` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useGetEventsQuery({
 *   variables: {
 *   },
 * });
 */
export function useGetEventsQuery(baseOptions?: Apollo.QueryHookOptions<GetEventsQuery, GetEventsQueryVariables>) {
        return Apollo.useQuery<GetEventsQuery, GetEventsQueryVariables>(GetEventsDocument, baseOptions);
      }
export function useGetEventsLazyQuery(baseOptions?: Apollo.LazyQueryHookOptions<GetEventsQuery, GetEventsQueryVariables>) {
          return Apollo.useLazyQuery<GetEventsQuery, GetEventsQueryVariables>(GetEventsDocument, baseOptions);
        }
export type GetEventsQueryHookResult = ReturnType<typeof useGetEventsQuery>;
export type GetEventsLazyQueryHookResult = ReturnType<typeof useGetEventsLazyQuery>;
export type GetEventsQueryResult = Apollo.QueryResult<GetEventsQuery, GetEventsQueryVariables>;

      export interface PossibleTypesResultData {
        possibleTypes: {
          [key: string]: string[]
        }
      }
      const result: PossibleTypesResultData = {
  "possibleTypes": {}
};
      export default result;
    