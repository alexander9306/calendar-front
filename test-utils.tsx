// This is a prepare dependecy test - that's why this rule is disable
/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import { render, RenderResult } from "@testing-library/react";
import { ApolloProvider } from "@apollo/client";
import { setupServer } from "msw/node";
import userEvent from "@testing-library/user-event";
import client from "./src/apollo-client/client";
import requestHandlers from "./request-handlers";

const AllTheProviders: React.FunctionComponent = ({ children }) => {
  return <ApolloProvider client={client}>{children}</ApolloProvider>;
};

// eslint-disable-next-line @typescript-eslint/no-explicit-any
type CustomRender = (ui: JSX.Element, options?: any) => RenderResult;

const customRender: CustomRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";

export { customRender as render };

const server = setupServer();

beforeAll(() => {
  server.listen({
    onUnhandledRequest: "error",
  });
  client.resetStore();
});
afterEach(() => {
  server.resetHandlers();
});

afterAll(() => {
  server.close();
  client.clearStore();
});

export { server, userEvent };
