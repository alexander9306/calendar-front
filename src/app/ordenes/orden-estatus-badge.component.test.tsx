import React from "react";
import { render, screen } from "test-utils";
import { create } from "react-test-renderer";
import OrdenStatusBadge from "./orden-estatus-badge.component";
import { OrdenEstatus } from "../generated/api";

describe("Orden Status Badge Component", () => {
  describe("Functionality Testing", () => {
    test("Verify badge with a pendiente text is rendered when number 0 is passed as an argument", () => {
      render(<OrdenStatusBadge estatus={OrdenEstatus.Pendiente} />);
      expect(screen.queryByText("pendiente")).toBeInTheDocument();
    });
    test("Verify badge with a pendiente text is rendered when a number different than 0 is passed as an argument", () => {
      render(<OrdenStatusBadge estatus={OrdenEstatus.Pendiente} />);
      expect(screen.queryByText("pendiente")).toBeNull();
    });
    test("Verify badge with a completado text is rendered when number 1 is passed as an argument", () => {
      render(<OrdenStatusBadge estatus={OrdenEstatus.Completada} />);
      expect(screen.queryByText("completado")).toBeInTheDocument();
    });
    test("Verify badge with a completado text is rendered when number 2 is passed as an argument", () => {
      render(<OrdenStatusBadge estatus={OrdenEstatus.Completada} />);
      expect(screen.queryByText("completado")).toBeInTheDocument();
    });
  });
  describe("Snapshot Testing", () => {
    test("Verify when a number 0 is passed as estatus prop", () => {
      const tree = create(
        <OrdenStatusBadge estatus={OrdenEstatus.Completada} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
    test("Verify when a number 1 is passed as estatus prop", () => {
      const tree = create(
        <OrdenStatusBadge estatus={OrdenEstatus.Pendiente} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
