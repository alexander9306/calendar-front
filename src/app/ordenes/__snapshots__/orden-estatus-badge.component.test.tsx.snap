// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orden Status Badge Component Snapshot Testing Verify when a number 0 is passed as estatus prop 1`] = `
<span
  className="badge badge-danger"
>
  pendiente
</span>
`;

exports[`Orden Status Badge Component Snapshot Testing Verify when a number 1 is passed as estatus prop 1`] = `
<span
  className="badge badge-success"
>
  completado
</span>
`;

exports[`Orden Status Badge Component Snapshot Testing Verify when no props are provided 1`] = `
<span
  className="badge badge-danger"
>
  pendiente
</span>
`;
